store/store.go

```diff
@@ -546,6 +559,110 @@ type requestOptions struct {
 	Data         []byte
 }
 
+type httpErrorResponse struct {
+	StatusCode int
+	URL        *url.URL
+	Method     string
+	Oops       string
+}
+
+func NewHttpErrorResponse(resp *http.Response) *httpErrorResponse {
+	return &httpErrorResponse{StatusCode: resp.StatusCode, URL: resp.Request.URL, Method: resp.Request.Method}
+}
+
+func (r httpErrorResponse) Error() string {
```

Comment:
Give this a pointer receiver, and drop the * from your calls to NewHttpErrorResponse